salt/modules/icinga2.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+'''
+Module to provide icinga2 compatibility to salt.
+
+:depends:   - icinga2 server
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import logging
+import os
+import subprocess
+
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load this module if the mysql libraries exist
+    '''
+    # TODO: icinga2 check should be os independent and maybe based on package, not binary.
+    if os.path.isfile('/usr/sbin/icinga2'):
+        return True
+    return (False, 'Icinga2 not installed.')
+
+
+def _execute(cmd, ret_code=False):
+    process = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    if ret_code:
+        return process.wait()
+    output, error = process.communicate()
+    if output:
+        log.debug(output)
+        return output
+    log.debug(error)
+    return error
+
+
+def generate_ticket(domain):
+    '''
+    Generate and save an icinga2 ticket.
+
+    Returns::
+        icinga2 pki ticket --cn domain.tld
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' icinga2.generate_ticket domain.tld
+
+    '''
+    result = _execute(["icinga2", "pki", "ticket", "--cn", domain])
+    return result
+
+
+def generate_cert(domain):
```

Comment:
With all of these cert commands, would it be prudent to allow a user to specify a path? Also, should we be checking for the validity of that path before calling out to icinga?